### PR TITLE
Task07 - Consume Real Data

### PR DESCRIPTION
**Pull request by nortega12**
# Pull Request
## Description
In this PR I added a js file where I put the json related functions: Fetch the data from the json file and Sort the array. Also removed the unused information from the html and added new functions in the homepage.js that adds new elements in the html and fill them with the new info.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
I downloaded the "Live Server" extension in Visual Studio Code so I could see the results while changing the code. You just have to download it, click in "Go Live" in the lower right corner and then it will automatically open a chrome window so you can see the results.

### CODE REVIEW

File: task07-consume-real-data/homepage.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ */
+import {fetchOnePieceData} from './json-data.js';
+
+"use strict";
+
+let OnePieceInformation;
+
+window.onload = async () => {
+    OnePieceInformation = await fetchOnePieceData();
+    console.log(OnePieceInformation);
```
Comment (kduranaraya) [inline task07-consume-real-data/homepage.js]: Please remove the console.logs

File: task07-consume-real-data/json-data.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    });
+}
+
+export async function fetchOnePieceData(){
+    const OnePieceUrl = 'https://static.akurey.com/trainings/OnePieceInformation.json';
+    
+    try {
+        let onePieceInfo = await fetch(OnePieceUrl)
+        .then( (response) => response.json())
+        .catch( (error) => console.log(error));
```
Comment (kduranaraya) [inline task07-consume-real-data/json-data.js]: Try to don't use console.log, in this case you can open a alert or something similar.

File: task07-consume-real-data/json-data.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+        sortArray(onePieceInfo.islands);
+        sortArray(onePieceInfo.characters);
+        sortArray(onePieceInfo.mysticObjects);
+
+        return await onePieceInfo;
+    } catch (error) {
+        console.log(error);
```
Comment (kduranaraya) [inline task07-consume-real-data/json-data.js]: Same here with the console.

File: task07-consume-real-data/json-data.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        return 0;
+    });
+}
+
+export async function fetchOnePieceData(){
+    const OnePieceUrl = 'https://static.akurey.com/trainings/OnePieceInformation.json';
+    
+    try {
+        let onePieceInfo = await fetch(OnePieceUrl)
+        .then( (response) => response.json())
```
Comment (kduranaraya) [inline task07-consume-real-data/json-data.js]: In this case if you use async/await you don't need to use the .then() and .catch() promise method. So you have a return in line 35 and also catch error in line 36, what is your purpose with those methods?

Comment (nortega12) [reply task07-consume-real-data/json-data.js]: Just trying to catch the error, but understood :) so i just leave the one of line 36 right?

File: task07-consume-real-data/json-data.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        return 0;
+    });
+}
+
+export async function fetchOnePieceData(){
+    const OnePieceUrl = 'https://static.akurey.com/trainings/OnePieceInformation.json';
+    
+    try {
+        let onePieceInfo = await fetch(OnePieceUrl)
+        .then( (response) => response.json());
```
Comment (kduranaraya) [inline task07-consume-real-data/json-data.js:28]: You can do something like:

try{
const onePieceData = await fetch(url);
const response = await onePieceData.json();
return response;
}

In this case you don't use the promises methods, just the await/async function :)

**Pull request merged**